### ISSUE TITLE
Loading cell templates with $.ajax cannot take advantage of $templateCache

### ISSUE THREAD

**Issue by kosprov** [angular-ui/ui-grid#422]
Loading external cell templates (cellTemplate, editableCellTemplate, headerCellTemplate) with $.ajax does not take advantage of $templateCache.

It would be useful to define complex templates in views using:

``` html
<script type="text/ng-template" id="myCellTemplate.html">
<!-- cell markup here -->
</script>
```

or in any other fashion that pre-loads them in the $templateCache (we are bundling views ala require.js/text and injecting them inside $templateCache during application bootstraping).

Using $http.get sounds like the way to go but due to the fact that it is asynchronous, it's a bit trick to refactor (I assume). 

Checking $templateCache before making the synchronous ajax call and updating it afterwards may do the job.

**Comment by dustinschultz** [comment]
Also bitten by this code. New browsers now emit: "Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check http://xhr.spec.whatwg.org/." due to the async:false code

**Comment by carlgieringer** [comment]
Why was this closed?

**Comment by rpocklin** [comment]
I'm not using this directive anymore, but I agree this shouldn't be closed.  My suggestion was just a workaround, in lieu of a real fix.

Will you accept a PR to fix this?

**Comment by PaulL1** [comment]
it looks to have been closed nearly a year ago because it was a quite old issue on 2.x, and everyone is focused on 3.0.  A PR might get accepted, but really there aren't any developers working on 2.x who could do the accpeting.

**Comment by wolf-daniel** [comment]
Our input on the subject:
After updating Chrome to version 43, we saw that ng-grid does not load the HTML templates (only for cellTemplate).
We came to understand that this happens due to lack of support to jQuery's 'async: false' approach.
We bypassed it by writing an AngularJS service in our app, that asynchronously loads all of the relevant templates and puts them in $templateCache. Applicatively, we make sure this happens at app's initialization, so that when a controller later uses ng-grid, the $templateCache would already be ready.
Works for us, for the meantime...

**Comment by lebolo** [comment]
Was this issue actually resolved in 3.0? I'm still seeing multiple HTTP requests for my custom templates.

Is the solution to explicitly cache custom templates upon module initialization (as above or as shown [here](https://thinkster.io/templatecache-tutorial/#/put_template))?
